tasks/setup.py

```diff
@@ -134,36 +135,24 @@ def setupAWSConfig(config: Config):
             break
         warn(f"{config.configParams.aws.account} is not a valid aws account")
 
-    # aws keypair name
-    if config.configParams.aws.keyPairName is None:
-        config.configParams.aws.keyPairName = getpass.getuser()
-    keyPairName = ask(f"🔑 Key pair name - stored in AWS Sandbox, default [{config.configParams.aws.keyPairName}]: ")
-    if len(keyPairName) > 0:
-        config.configParams.aws.keyPairName = keyPairName
+    # ask user if they want to create a new key or import an existing key
+    if ask_yesno("Do you want to create a new key pair?"):
+        _aws_create_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
+    elif ask_yesno("Do you want to import an existing key pair?"):
+        _aws_import_keypair(ctx, config, use_aws_vault=True, aws_account_name=config.configParams.aws.account)
+    else:
+        warn("No key pair configured, you will need to manually configure a key pair")
 
     # check keypair name
     if config.options is None:
         config.options = Config.Options(checkKeyPair=False)
     default_check_key_pair = "Y" if config.options.checkKeyPair else "N"
     checkKeyPair = ask(
-        f"Did you create your SSH key on AWS and want me to check it is loaded on your ssh agent when creating manual environments or running e2e tests [Y/N]? Default [{default_check_key_pair}]: "
+        f"Do you want check that the keypair is loaded in ssh agent when creating manual environments or running e2e tests [Y/N]? Default [{default_check_key_pair}]: "
     )
     if len(checkKeyPair) > 0:
         config.options.checkKeyPair = checkKeyPair.lower() == "y" or checkKeyPair.lower() == "yes"
 
-    # aws public key path
-    if config.configParams.aws.publicKeyPath is None:
-        config.configParams.aws.publicKeyPath = str(Path.home().joinpath(".ssh", "id_ed25519.pub").absolute())
-    default_public_key_path = config.configParams.aws.publicKeyPath
-    while True:
-        config.configParams.aws.publicKeyPath = default_public_key_path
-        publicKeyPath = ask(f"🔑 Path to your public ssh key, default [{config.configParams.aws.publicKeyPath}]: ")
-        if len(publicKeyPath) > 0:
-            config.configParams.aws.publicKeyPath = publicKeyPath
-        if os.path.isfile(config.configParams.aws.publicKeyPath):
-            break
-        warn(f"{config.configParams.aws.publicKeyPath} is not a valid ssh key")
-
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Removing this code might break the experience for folks with an already configured environment, we should try to be backward compatible

Reply:
it wasn't removed, it was moved to [resolve_keypair_opts](https://github.com/DataDog/test-infra-definitions/pull/1045/files#diff-c17ad71d631afbbe74eea29844a37a1501410870263f1ea09d58f4a786e91772R279-R283).